base/complex.jl

```diff
@@ -107,7 +107,7 @@ abs(z::Complex)  = hypot(real(z), imag(z))
 abs2(z::Complex) = real(z)*real(z) + imag(z)*imag(z)
 inv(z::Complex)  = conj(z)/abs2(z)
 inv{T<:Integer}(z::Complex{T}) = inv(float(z))
-sign(z::Complex) = z/abs(z)
+sign(z::Complex) = z == 0? float(zero(z)) : z/abs(z)
```

Comment:
Why not just use `z::Number` here?   There's no reason this definition could not work for quaternions, for example.  And put it in `number.jl`.